win32ss/user/user32/windows/dialog.c

```diff
@@ -2624,7 +2624,17 @@ IsDialogMessageW(
              if (!(dlgCode & DLGC_WANTARROWS))
              {
                  BOOL fPrevious = (lpMsg->wParam == VK_LEFT || lpMsg->wParam == VK_UP);
-                 HWND hwndNext = GetNextDlgGroupItem( hDlg, lpMsg->hwnd, fPrevious );
+                 HWND hwndFirst = lpMsg->hwnd;
+                 HWND hwndNext = GetNextDlgGroupItem(hDlg, hwndFirst, fPrevious);
+
+                 /* Skip STATIC elements when arrow-moving through a list of controls */
+                 while (hwndNext && (SendMessageW(hwndNext, WM_GETDLGCODE, 0, 0) & DLGC_STATIC))
+                 {
+                     if (hwndFirst == hwndNext)
+                         return TRUE;
```

Comment:
"return" ? What about the code that is below, outside the while-loop? Shouldn't it run the WM_NEXTDLGCTL from below?

Reply:
If they are equal, then I think there is no need to get the next.